src/libxively/xi_debug.c

```diff
@@ -37,3 +38,29 @@ void xi_debug_data_logger_impl( const char* msg, const xi_data_desc_t* data_desc
 
     xi_free( tmp );
 }
+
+/** Trims the absolute file path to just the file name */
+const char* xi_debug_dont_print_the_path( const char* msg )
```

Comment:
I don't think any special compiler flag is required. It is a macro defined by -D . The make system fills it with proper content: filename